internal/ai/ai.go

```diff
@@ -66,8 +69,29 @@ func (c *CachingAI) Chat() (string, error) {
 		}
 	}
 
-	response, err := c.ai.Chat()
-	if err != nil {
+	var response string
+	var err error
+	var lastErr error
+	retries := 0
+	maxRetries := c.config.MaxRetries
+
+	for retries <= maxRetries {
+		if retries > 0 {
+			if c.config.RetryBackoff > 0 {
+				backoffTime := time.Duration(c.config.RetryBackoff * float64(time.Second))
+				log.Infof("Retrying AI request in %.2f seconds (attempt %d/%d)", c.config.RetryBackoff, retries, maxRetries)
+				time.Sleep(backoffTime)
```

Comment:
then this becomes something like:

https://github.com/blacktop/ipsw/blob/10534629ea77e881124d6ed2dcbbdfbf4d5e9473/cmd/ipsw/cmd/watch.go#L566